Feature/issue 22/implement autowired bean post processor

olehhaliak opened the pull request:
## Describe your changes

## Issue #22 Implement autowired bean post processor

## Checklist before requesting a review
- [x] There are no MR remarks
- [x] Clean and concise code that is easy to read and understand
- [x] Code covered with tests and they are green
- [x] Code API has informative JavaDoc on it
- [x] Rich logging with detailed messages and correct use of log levels

olehh-amway committed:
[issue-#22] Implement AutowiredBeanPostProcessor

olehh-amway committed:
[issue-#22] Implement DefaultDependencyResolver

olehh-amway committed:
[issue-#22] Added logging

vitshch commented on src/main/java/com/bringframework/factory/AutowiredBeanPostProcessor.java:
```diff
@@ -0,0 +1,71 @@
+    definitionRegistry = registry;
+    dependencyResolver = new DefaultDependencyResolver(definitionRegistry.getAllBeanDefinitions());
+  }
+
```
Extra empty line

olehhaliak replied:
fixed

olehh-amway committed:
Merge branch 'dev' of github.com:bobocode-blyznytsia/bring-framework into feature/issue-22/implement-autowired-bean-post-processor

olehh-amway committed:
[issue-#22] Fixed Autowired BPP and tests after pulling changes from dev

olehh-amway committed:
[issue-#22] Add new line at the end of files

marynamelnichuk commented on src/main/java/com/bringframework/exceptions/BeanInjectionException.java:
```diff
@@ -0,0 +1,8 @@
+package com.bringframework.exceptions;
+
```
Please add documentation here, cause for all exceptions classes we have java doc

olehhaliak replied:
fixed

marynamelnichuk commented on src/test/java/com/bringframework/factory/AutowiredBeanPostProcessorTest.java:
```diff
@@ -0,0 +1,75 @@
+
+    autowiredBPP.process();
+
+    Assertions.assertSame(testClassABean, secondBean.testClassA);
```
Use static import here for consistency

olehhaliak replied:
Fixed

marynamelnichuk commented on src/test/java/com/bringframework/resolver/DefaultDependencyResolverTest.java:
```diff
@@ -0,0 +1,97 @@
+
+  @Test
+  void candidateResolvedForSameType() {
+    var linkedListBeanDefinition = mock(BeanDefinition.class);
+    doReturn(LinkedList.class).when(linkedListBeanDefinition).getBeanClass();
+
+    var expectedCandidateName = "linkedList";
+    Map<String, BeanDefinition> beanDefinitionMap =
+        Map.of(expectedCandidateName, linkedListBeanDefinition);
+
+    var dependencyResolver = new DefaultDependencyResolver(beanDefinitionMap);
```
These snippet is repeated at least 3 times, please create a helper method with this logic.

olehhaliak replied:
consolidated this logic in mockBeanDefinitionOfClass() and linkedListBeanDefinitionMap() methods

marynamelnichuk commented on src/test/java/com/bringframework/factory/AutowiredBeanPostProcessorTest.java:
```diff
@@ -0,0 +1,75 @@
+class AutowiredBeanPostProcessorTest {
+
+  @Mock
+  DependencyResolver dependencyResolver;
```
Please add here private modifier and in other lines where it is needed

olehhaliak replied:
done

marynamelnichuk commented on src/main/java/com/bringframework/factory/AutowiredBeanPostProcessor.java:
```diff
@@ -0,0 +1,62 @@
+  private final DependencyResolver dependencyResolver;
+
+  public AutowiredBeanPostProcessor(BeanDefinitionRegistry registry, Map<String, Object> rawBeans) {
+    this.rawBeans = rawBeans;
```
Please follow the same style in your code, if you use "this" to set fields in constructor, make the same for all fields.

olehhaliak replied:
fixed

marynamelnichuk commented on src/test/java/com/bringframework/resolver/DefaultDependencyResolverTest.java:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  @Test
+  void multipleCandidateFound_NoUniqueBeanExceptionThrown() {
```
I don't remember whether we agreed on such test methods naming.

vitshch replied:
No, we don't, but I think we have to use **camelCase** in tests naming

olehhaliak replied:
renamed in camelCase and added `@DisplayName` instead to preserve readability.

vitshch commented on src/main/java/com/bringframework/factory/AutowiredBeanPostProcessor.java:
```diff
@@ -0,0 +1,62 @@
+    beanDefinition.getAutowiredFieldsMetadata().values()
+        .forEach(field -> populateField(field, bean));
+  }
+
```
Remove empty line

olehhaliak replied:
fixed

olehh-amway committed:
[issue-#22] Fixed comments

- added docs for BeanInjectionException
- unified usage of 'this.' in AutowiredBPP constructor
- added private modifiers on fields in AutowiredBPPTest
- added missing static import
- changed test namings
- reorganized boilerplate code in tests

olehh-amway committed:
[issue-#22] Increased max line length to 120

olehh-amway committed:
[issue-#22] fixed spacings

olehh-amway committed:
[issue-#22] constants made private

olehh-amway committed:
[issue-#22] added @Display name for particularly long test names
